tests/unit/relations/test_db.py

```diff
@@ -26,40 +30,59 @@ def setUp(self):
         self.harness.begin()
 
         self.charm = self.harness.charm
+        self.app = self.charm.app.name
+        self.unit = self.charm.unit.name
+        self.backend = self.charm.backend
         self.db_relation = self.charm.legacy_db_relation
         self.db_admin_relation = self.charm.legacy_db_admin_relation
 
+        # TODO update mocks now we're using the mock harness
+        # Define a backend relation
+        self.backend_rel_id = self.harness.add_relation(BACKEND_RELATION_NAME, "postgres")
+        self.harness.add_relation_unit(self.backend_rel_id, "postgres/0")
+        self.harness.add_relation_unit(self.backend_rel_id, self.unit)
+
+        # Define a db relation
+        self.db_rel_id = self.harness.add_relation(DB_RELATION_NAME, "postgres")
+        self.harness.add_relation_unit(self.db_rel_id, "client/0")
+        self.harness.add_relation_unit(self.db_rel_id, self.unit)
+
+        # Define a db-admin relation
+        self.db_admin_rel_id = self.harness.add_relation(DB_ADMIN_RELATION_NAME, "postgres")
+        self.harness.add_relation_unit(self.db_admin_rel_id, "admin_client/0")
+        self.harness.add_relation_unit(self.db_admin_rel_id, self.unit)
+
```

Comment:
You can change `postgres`to another application name here in both cases. I think it makes more sense.